spring-integration-hazelcast/src/test/java/org/springframework/integration/hazelcast/inbound/HazelcastClusterMonitorInboundChannelAdapterTests.java

```diff
@@ -0,0 +1,228 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.inbound;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.config.GroupConfig;
+import com.hazelcast.core.Client;
+import com.hazelcast.core.ClientType;
+import com.hazelcast.core.DistributedObjectEvent;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.LifecycleEvent;
+import com.hazelcast.core.Member;
+import com.hazelcast.core.LifecycleEvent.LifecycleState;
+import com.hazelcast.core.MembershipEvent;
+import com.hazelcast.core.MigrationEvent;
+
+/**
+ * Hazelcast Cluster Monitor Inbound Channel Adapter Unit Test Class
+ *
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+@DirtiesContext
+public class HazelcastClusterMonitorInboundChannelAdapterTests {
+
+	private static final String TEST_GROUP_NAME1 = "Test_Group_Name1";
+
+	private static final int TIMEOUT = 10_000;
+
+	@Autowired
+	private PollableChannel cmChannel1;
+
+	@Autowired
+	private PollableChannel cmChannel2;
+
+	@Autowired
+	private PollableChannel cmChannel3;
+
+	@Autowired
+	private PollableChannel cmChannel4;
+
+	@Autowired
+	private PollableChannel cmChannel5;
+
+	@Autowired
+	private PollableChannel cmChannel6;
+
+	@Autowired
+	private HazelcastInstance hazelcastInstance;
+
+	@Autowired
+	private HazelcastInstance hazelcastInstance2;
+
+	@Autowired
+	private HazelcastInstance hazelcastInstance3;
+
+	@Test
+	public void testMembershipEvent() {
+		testMembershipEvent(hazelcastInstance, cmChannel1, "testKey1", "testValue1");
+	}
+
+	@Test
+	public void testDistributedObjectEvent() {
+		testDistributedObjectEventByChannelAndHazelcastInstance(cmChannel2,
+				hazelcastInstance);
+	}
+
+	@Test
+	public void testMigrationEvent() {
+		hazelcastInstance3.getLifecycleService().terminate();
+		final Message<?> msg = cmChannel3.receive(TIMEOUT);
+		verifyMigrationEvent(msg);
+	}
+
+	@Test
+	public void testLifecycleEvent() throws InterruptedException {
```

Comment:
This test pass, but I see the WARN in logs:
```
2015-07-13 11:59:31,316 WARN DefaultLifecycleProcessor [main] : Failed to stop bean 'org.springframework.integration.hazelcast.inbound.HazelcastClusterMonitorMessageProducer#3'
com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not active!
	at com.hazelcast.instance.TerminatedLifecycleService.removeLifecycleListener(TerminatedLifecycleService.java:45)
	at org.springframework.integration.hazelcast.inbound.HazelcastClusterMonitorMessageProducer.doStop(HazelcastClusterMonitorMessageProducer.java:123)
	at org.springframework.integration.endpoint.AbstractEndpoint.doStop(AbstractEndpoint.java:145)
```
I think we should check `hazelcastInstance` state in the `HazelcastClusterMonitorMessageProducer` before removal any listener.